lib/EarlyFunctionAnalysis/EarlyFunctionAnalysis.cpp

```diff
@@ -1828,9 +1859,16 @@ bool EarlyFunctionAnalysis::runOnModule(Module &M) {
   for (const auto &[_, Node] : BasicBlockNodeMap)
     RootNode->addSuccessor(Node);
 
+  // The intraprocedural analysis will be scheduled only for those functions
+  // which have `Invalid` and `Fake` as type.
   UniquedQueue<BasicBlockNode *> CFEPQueue;
   for (auto *Node : llvm::post_order(&CG)) {
-    if (Node != RootNode)
+    if (Node == RootNode)
+      continue;
+
+    auto &Function = Binary.Functions[getBasicBlockPC(Node->BB)];
+    if (Function.Type == model::FunctionType::Invalid
+        || Function.Type == model::FunctionType::Fake)
```

Comment:
As I mentioned, I think we need to ensure that `Fake` functions are analyzed first.
Also, we probably don't need to analyze them, we could find out that they are not fake, but we don't want that.
Basically, we just want to obtain an `llvm::Function` we can inline. This might to reorganize some stuff.

Reply:
While we basically only want the `llvm::Function`, I thought it was not too much of a problem if fake functions are wholly recomputed too. But it's true, we need to make sure they are recomputed first. Initially I kinda thought that this was what was actually happening since fake functions should be leaf functions too (?), but I guess we want to make completely sure that fake functions are pushed onto the queue first.

Reply:
If you do a full blown analysis, EFA could determine that it's not fake and that's not what we want.

Reply:
OK, trying to see if we can extract the code responsible for creating the fake function for the `llvm::Function*`.

Reply:
Fix this by introducing `CFEPAnalyzer::initializeFakeFunctions` which re-create an `llvm::Function *` for fake functions; no need to put them onto the queue / ensure they're analyzed first at this point. I think it should be OK now.

Reply:
That sounds nice!